src/pages/RequestCallPage.js

```diff
@@ -162,13 +171,30 @@ class RequestCallPage extends Component {
     }
 
     validatePhoneInput() {
-        if (_.isEmpty(this.state.phoneNumber.trim())) {
-            this.setState({phoneNumberError: this.props.translate('messages.noPhoneNumber')});
-        } else if (!Str.isValidPhone(this.state.phoneNumber)) {
-            this.setState({phoneNumberError: this.props.translate('messages.errorMessageInvalidPhone')});
-        } else {
-            this.setState({phoneNumberError: ''});
+        this.setState((prevState) => ({phoneNumberError: this.getPhoneNumberError(prevState.phoneNumber)}));
+    }
+
+    /**
+     * Checks for input errors, returns true if everything is valid, false otherwise.
+     * @returns {Boolean}
+     */
+    validateInputs() {
+        const firstOrLastNameEmpty = _.isEmpty(this.state.firstName.trim()) || _.isEmpty(this.state.lastName.trim());
+        if (firstOrLastNameEmpty) {
+            Growl.error(this.props.translate('requestCallPage.growlMessageEmptyName'));
         }
+
+        this.setState((prevState) => {
+            const phoneNumberError = this.getPhoneNumberError(prevState.phoneNumber);
+            const nameErrors = getFirstAndLastNameErrors(prevState.firstName, prevState.lastName);
```

Comment:
Looks like we just need to initialize these outside the `setState` call

Reply:
wait i may be confused lol

Reply:
No I don't think I'm confused anymore. `setState` is async so I'm trying to figure out how to get the timing right haha

Reply:
Huh, turns out a bit of destructuring cured or confused eslint haha